index.js

```diff
@@ -125,6 +125,20 @@ const getCountriesByFamousFor = (famousThing) => {
   );
 };
 
+/** 
+ * Returns an array of objects containing all countries, each containing `country`, `capital`,
+ * `currency`, `native_language`, `famous_for`, `phone_code`, `flag` and `drive_direction` filtered by `is_landlocked`
+ * @param {'true' | 'false'} isLandLocked  Country is said to be landlocked when it is surrounded on all sides by one or more countries
+ * @returns {Array} An array of country objects
+*/
+const getCountriesByLandLock = (isLandLocked) => {
+  const isBoolean = typeof isLandLocked === 'boolean'
+  const filterBy = isBoolean && isLandLocked
```

Comment:
I am not sure if it is so necessary to have that kind of checks. I think that proper documentation and `@param` Definition should be enough for the _API consumers_.

Reply:
Just making sure that user only enters boolean, but i get your point the definition should be enough.
But in order to `through an error` i need to check type of the value.

Reply:
If the user enters any value other than boolean, i will throw an error suggesting proper value.